### PR TITLE
Feature/#396 unique and more names

### LINKED ISSUE #396: Citizen names should be unique
We have more then enough name combinations already to achieve this.

Each citizens name should be unique in his colony. So on spawn we check all other citizens and discard names that already exist. Since we have a <1% hit rate, this runs in Ω(2).

We should also allow admins/owners to specifiy citizen commands via the citizens name (now that that is unique) and offer autocompletion for it

### PR DESCRIPTION
**Pull request by nojo121**
Closes #396 

# Changes proposed in this pull request:
-Citizen names are now unique
-now there are 136 male names and 136 female names
Review please

### CODE REVIEW

**Comment by marvin-bitterlich** [comment]
LGTM

[Approved with PullApprove](https://pullapprove.com/Minecolonies/minecolonies/pull-request/423/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=Minecolonies/minecolonies)

**Comment by Raycoms** [comment]
:shipit: I guess.

[Approved with PullApprove](https://pullapprove.com/Minecolonies/minecolonies/pull-request/423/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=Minecolonies/minecolonies)

File: src/main/java/com/minecolonies/coremod/colony/CitizenData.java [hunk 1]
```diff
@@ -241,16 +241,23 @@ public void initializeFromEntity(@NotNull final EntityCitizen entity)
      */
     private String generateName(@NotNull final Random rand)
     {
-        final String firstName;
-        if (!female)
+        String citizenName;
+        if (female)
         {
-            firstName = getRandomElement(rand, Configurations.maleFirstNames);
+            citizenName = String.format("%s %s. %s", getRandomElement(rand, Configurations.femaleFirstNames), getRandomLetter(rand), getRandomElement(rand, Configurations.lastNames));
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/colony/CitizenData.java:247]: MINOR Split this 183 characters long line (which is greater than 180 authorized). [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS00103)

File: src/main/java/com/minecolonies/coremod/colony/CitizenData.java [hunk 2]
```diff
@@ -241,16 +241,23 @@ public void initializeFromEntity(@NotNull final EntityCitizen entity)
...
         }
         else
         {
-            firstName = getRandomElement(rand, Configurations.femaleFirstNames);
+            citizenName = String.format("%s %s. %s", getRandomElement(rand, Configurations.maleFirstNames), getRandomLetter(rand), getRandomElement(rand, Configurations.lastNames));
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/colony/CitizenData.java:251]: MINOR Split this 181 characters long line (which is greater than 180 authorized). [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS00103)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 2 issues
* MINOR 2 minor

Watch the comments in this conversation to review them.

**Pull request merged**